File: solutions/2026-02-17/java/2026-02-17-binary-watch.java
```java
import java.util.ArrayList;
import java.util.List;

// Problem: Binary Watch
// Link: https://leetcode.com/problems/binary-watch/
// Approach:
// We are given the total number of turned-on LEDs (turnedOn) and need to find all possible times (hours and minutes)
// that satisfy this condition. The hours are represented by 4 LEDs and minutes by 6 LEDs.
//
// We can iterate through all possible hour values (0-11) and for each hour, determine how many LEDs are on.
// If the number of LEDs on for the hour is `h_on`, then the remaining `turnedOn - h_on` LEDs must be on for the minutes.
// We then iterate through all possible minute values (0-59). For each minute, we count the number of set bits.
// If the number of set bits for the minute is equal to `turnedOn - h_on`, then we have found a valid time.
//
// A more efficient approach is to use backtracking or recursion.
// We can think of this as selecting `h_on` LEDs out of 4 for the hour and `m_on` LEDs out of 6 for the minute,
// such that `h_on + m_on = turnedOn`.
//
// We can define two recursive functions: one for hours and one for minutes.
//
// `generateHours(index, count, currentHour, turnedOn, result)`:
//   - `index`: the current LED position (0-3) for the hour.
//   - `count`: the number of LEDs turned on so far for the hour.
//   - `currentHour`: the current hour value being built.
//   - `turnedOn`: the total number of LEDs that need to be turned on.
//   - `result`: the list to store valid time strings.
//
// `generateMinutes(index, count, currentMinute, turnedOn, result)`:
//   - `index`: the current LED position (0-5) for the minute.
//   - `count`: the number of LEDs turned on so far for the minute.
//   - `currentMinute`: the current minute value being built.
//   - `turnedOn`: the total number of LEDs that need to be turned on for the minutes.
//   - `result`: the list to store valid time strings.
//
// Base cases:
// For `generateHours`: If `index == 4`, we have considered all hour LEDs. If `count <= turnedOn`, we then call `generateMinutes`
// to find the corresponding minute combinations.
// For `generateMinutes`: If `index == 6`, we have considered all minute LEDs. If `count == turnedOn`, we format and add the time to the result.
//
// Recursive steps:
// For each `index`, we have two choices:
// 1. Turn the LED off: Recursively call with `index + 1`, `count`, `currentHour` (or `currentMinute`).
// 2. Turn the LED on: If `count < turnedOn`, recursively call with `index + 1`, `count + 1`, and update `currentHour`
//    (by adding `1 << index`) or `currentMinute` (by adding `1 << index`).
//
// Constraints:
// - The hour value must be between 0 and 11.
// - The minute value must be between 0 and 59.
// - The hour should not have a leading zero unless it's 0 itself (e.g., "0:30" is valid, "01:00" is not).
// - The minute must be two digits (e.g., "10:02" is valid, "10:2" is not).
//
// We can optimize this by iterating through all possible splits of `turnedOn` into `h_on` (hours LEDs on) and `m_on` (minutes LEDs on).
// For each `h_on` from 0 to `turnedOn` (and `h_on <= 4`), calculate `m_on = turnedOn - h_on`.
// If `m_on` is valid (0 <= `m_on` <= 6), then we find all hours with `h_on` LEDs on and all minutes with `m_on` LEDs on.
//
// Function `countSetBits(n)`: Helper to count set bits in an integer.
//
// Time Complexity:
// The maximum value for `turnedOn` is 10.
// For each `turnedOn` value, we iterate through possible splits of `h_on` and `m_on`.
// For each split, we generate combinations of hours and minutes.
// The number of ways to choose `k` bits out of `n` is given by the binomial coefficient C(n, k).
// Maximum C(4, 4) * C(6, 6) is manageable.
// A more direct analysis: The outer loops for `h_on` and `m_on` run at most `turnedOn + 1` times.
// Inside these loops, we iterate through all possible hours (12) and minutes (60).
// For each hour/minute, we count bits, which takes O(log N) or O(1) if we precompute or use built-in functions.
// The total number of valid times is relatively small.
// A loose upper bound might seem high, but the constraints on `turnedOn` and the hour/minute limits keep it feasible.
// The actual complexity is closer to O(1) in practice due to small input ranges and limited output.
//
// Space Complexity:
// O(1) if we don't consider the output list.
// If we consider the output list, the space complexity depends on the number of possible valid times, which is bounded.
// In the worst case, the number of possible times can be up to around 70.
//
class Solution {
    /**
     * Generates all possible times for a binary watch given the number of turned-on LEDs.
     *
     * @param turnedOn The total number of LEDs that are currently on.
     * @return A list of all possible time strings in "H:MM" format.
     */
    public List<String> readBinaryWatch(int turnedOn) {
        List<String> result = new ArrayList<>();

        // Iterate through all possible hour values (0-11).
        for (int h = 0; h < 12; h++) {
            // Iterate through all possible minute values (0-59).
            for (int m = 0; m < 60; m++) {
                // Count the number of set bits (turned-on LEDs) for the current hour and minute.
                // The hour uses 4 bits and the minute uses 6 bits.
                // Integer.bitCount(n) returns the number of set bits in the binary representation of n.
                if (Integer.bitCount(h) + Integer.bitCount(m) == turnedOn) {
                    // Format the time string.
                    // %d for hour (no leading zero if h > 0, handled by String.format).
                    // %02d for minute (ensures two digits with a leading zero if necessary).
                    result.add(String.format("%d:%02d", h, m));
                }
            }
        }
        return result;
    }
}
```

File: solutions/2026-02-17/python/2026-02-17-binary-watch.py
```python
# Summary: Given an integer turnedOn representing the number of LEDs on a binary watch,
# find all possible valid times (hours and minutes) that can be displayed.
# Link: https://leetcode.com/problems/binary-watch/
# Approach:
# This problem can be solved using a backtracking or brute-force approach with bit manipulation.
# We can iterate through all possible hours (0-11) and all possible minutes (0-59).
# For each hour and minute combination, we can count the number of set bits (LEDs turned on).
# If the total number of set bits (from hour and minute) equals turnedOn, we format the time and add it to the result list.
#
# To count set bits efficiently, we can use the `bin()` function in Python and count the '1's, or use bitwise operations.
# Alternatively, we can use a recursive helper function (backtracking) to explore combinations of turning on a certain number of LEDs for hours and minutes separately.
#
# For the brute-force approach (chosen here for simplicity and clarity given the constraints):
# 1. Iterate through hours from 0 to 11.
# 2. For each hour, count the number of set bits. Let this be `hour_bits`.
# 3. If `hour_bits` is greater than `turnedOn`, continue to the next hour (no need to check minutes).
# 4. Calculate the remaining LEDs to be turned on for minutes: `minute_bits_needed = turnedOn - hour_bits`.
# 5. Iterate through minutes from 0 to 59.
# 6. For each minute, count the number of set bits. Let this be `minute_bits`.
# 7. If `minute_bits` equals `minute_bits_needed`, format the hour and minute into a string "H:MM" and add it to the result list.
#
# Formatting the time:
# - Hour should not have a leading zero unless it's 0.
# - Minute must be two digits, with a leading zero if necessary.
#
# Time Complexity: O(12 * 60 * log(max_value)) where max_value is the maximum bit representation for hour (11) or minute (59).
# The bit counting takes logarithmic time relative to the number. Since the number of hours and minutes are constant (12 and 60), this is effectively O(1).
# Space Complexity: O(N) where N is the number of valid times returned. In the worst case, all combinations might be valid, but given the constraints, the number of results is manageable.
# The maximum number of set bits is 10. The hour has 4 bits and the minute has 6 bits.
# Max possible set bits for hour: 4 (for 7, 11).
# Max possible set bits for minute: 6 (for 63, but we only go up to 59). Max for 59 is 5 bits.
# Max total set bits from hour and minute is effectively limited by the input `turnedOn`.
class Solution:
    def readBinaryWatch(self, turnedOn: int) -> list[str]:
        # List to store the resulting times
        result = []

        # Iterate through all possible hours from 0 to 11
        for h in range(12):
            # Count the number of set bits (LEDs on) in the current hour
            # bin(h) returns a string like '0b101'. We slice from index 2 to remove '0b'.
            # Then, we count the occurrences of '1'.
            hour_bits = bin(h)[2:].count('1')

            # If the number of set bits in the hour already exceeds turnedOn,
            # then this hour cannot be part of a valid time. Skip to the next hour.
            if hour_bits > turnedOn:
                continue

            # Calculate how many LEDs need to be on for the minutes
            minute_bits_needed = turnedOn - hour_bits

            # Iterate through all possible minutes from 0 to 59
            for m in range(60):
                # Count the number of set bits (LEDs on) in the current minute
                minute_bits = bin(m)[2:].count('1')

                # If the number of set bits in the minute matches the required number,
                # then this hour and minute combination forms a valid time.
                if minute_bits == minute_bits_needed:
                    # Format the time string.
                    # Hour 'h' is formatted without a leading zero unless it's 0.
                    # Minute 'm' is formatted to always have two digits, with a leading zero if necessary (e.g., "01", "09").
                    result.append(f"{h}:{m:02d}")

        # Return the list of all valid times
        return result

```